### PR TITLE
Example use regarding how to read and save images from E57 files

### PR DESCRIPTION
**Pull request by bigeyesung**
Hi,
I am adding this to read/save panoramic images from E57 files.
Also, I think it could be convenient to create a "example.cpp" that shows how to use the class/functions.
Let me know if you have any questions.

### CODE REVIEW

**Comment by asmaloney** [comment]
Thank you @bigeyesung.

Some initial things to fix:

1. The PR is not passing CI. When you set up the build for the example executable in cmake, you need to make it use C++11.
2. I would prefer putting the example (and its cmake build instructions) in a new folder - _examples_ - and changing the name: _examples/src/ImageExample.cpp_ or something. That would make it easier to add other examples in the future.
3. The example should not be built by default. If the developer is including E57Format as a submodule it should not be built. If it is being built on its own, it should be an option `E57_BUILD_EXAMPLES` (default to false). 
4. You need to run clang-format on the code for proper formatting (see README -> Contributing).

**Comment by bigeyesung** [comment]
> Thank you @bigeyesung.
> 
> Some initial things to fix:
> 
> 1. The PR is not passing CI. When you set up the build for the example executable in cmake, you need to make it use C++11.
> 2. I would prefer putting the example (and its cmake build instructions) in a new folder - _examples_ - and changing the name: _examples/src/ImageExample.cpp_ or something. That would make it easier to add other examples in the future.
> 3. The example should not be built by default. If the developer is including E57Format as a submodule it should not be built. If it is being built on its own, it should be an option `E57_BUILD_EXAMPLES` (default to false).
> 4. You need to run clang-format on the code for proper formatting (see README -> Contributing).

@asmaloney, thanks for the feedback, I am working on it.
I have one question in terms of #4 "clang-format verification". I am trying to make my "ReadImages.cpp" to run it automatically, not sure how to do it. I traced code to cmake/Modules/ClangFormat.cmake,  
``
if ( E57_CLANG_FORMAT_EXE )  
     get_target_property( e57_sources ${PROJECT_NAME} SOURCES )
``
Currently it checks include and src folders. Not sure how to make it to scan other folders (E.g. examples dir for this case).
I tried using filters and checked cmake doc. Any hints is appreciated. thx
``  
list( FILTER e57_sources INCLUDE REGEX ".*/examples/.*" )  
``  
https://cmake.org/cmake/help/latest/manual/cmake-properties.7.html#target-properties

File: examples/src/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# SPDX-License-Identifier: MIT
+# Copyright 2020 Andy Maloney <asmaloney@gmail.com>
+
+project(Examples)
+
+set( APP_EXE ReadImages )
+set(CMAKE_CXX_STANDARD 11)
```
Comment (asmaloney) [inline examples/src/CMakeLists.txt:7]: These should be set on the target, not globally. See `set_target_properties` in the main cmake file.

File: examples/src/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+add_executable(${APP_EXE} ReadImages.cpp)
```
Comment (asmaloney) [inline examples/src/CMakeLists.txt:9]: Sources should be added with `target_sources`. See src/CMakeLists.txt.

File: examples/src/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+target_link_libraries(${APP_EXE} PUBLIC E57Format)
```
Comment (asmaloney) [inline examples/src/CMakeLists.txt:10]: Should probably be **PRIVATE**.

File: examples/src/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# SPDX-License-Identifier: MIT
+# Copyright 2020 Andy Maloney <asmaloney@gmail.com>
+
+project(Examples)
```
Comment (asmaloney) [inline examples/src/CMakeLists.txt:4]: Should not declare a _project_ here.

File: examples/src/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+
+set( APP_EXE ReadImages )
```
Comment (asmaloney) [inline examples/src/CMakeLists.txt:6]: Not sure using a variable for this is necessary.

If you're going to use one, please name it something descriptive like `IMAGES_EXAMPLE_TARGET`.

File: examples/src/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# SPDX-License-Identifier: MIT
```
Comment (asmaloney) [inline examples/src/CMakeLists.txt:1]: This file should be up one level in the `examples` directory, not in the `example/src` directory.

**Review by asmaloney — CHANGES_REQUESTED** [review]

**Comment by asmaloney** [comment]
> I have one question in terms of #4 "clang-format verification"

So everything is done in terms of `targets` to keep as much as possible out of the global scope. In the clang-format file, I'm looking at the E57Format project, grabbing the sources files, filtering some out, and running clang-format on the list of files.

Since this file is included after your new example, you should be able to mimic this using your target.

So before `add_custom_target`, try grabbing the source files from your target using `get_target_property` with your target name(`ReadImages`), then adding them to the `e57_sources` list. (All wrapped in `if(E57_BUILD_EXAMPLES)` of course.)

**Review by bigeyesung — COMMENTED** [review]
working progress

**Comment by asmaloney** [comment]
Seems to have been abandoned. If not, please split into two PRs - one for the change to the lib, one for the example - and apply all the changes above (including formatting).

**Pull request closed without merging**